core/match.go

```diff
@@ -157,7 +158,7 @@ func UpdateDirsPermissionsRW(dir string) {
 		if f.IsDir() {
```

Comment:
The cleaner way to do this is -- 

if !f.IsDir() {
      return nil
}
err := os.Chmod(path.0700)
if (err != nil) {
log.Error(...)
}

Also, why are we doing a write operation as a part of our scanning process ?